### PR TITLE
Update SB,Storage,KeyVault SDK to T2

### PR DESCRIPTION
**Pull request by wantedfast**
**SDK Updated List**

1. Service BUS SDK
2. Azure Key Vault
3. Storage Account

### CODE REVIEW

File: Program.cs [hunk 1]
```diff
@@ -161,7 +163,7 @@ public static void Main(string[] args)
                 // Authenticate
                 var credentials = SdkContext.AzureCredentialsFactory.FromFile(Environment.GetEnvironmentVariable("AZURE_AUTH_LOCATION"));
 
-                var azure = Azure
+                var azure = Microsoft.Azure.Management.Fluent.Azure
```
Comment (wantedfast) [inline Program.cs:166]: Because `Azure` exist both in management SDK and T2 data SDK that It need specify it's namespace.

Comment (tzhanl) [reply Program.cs:166]: @jongio Please review.

Comment (jongio) [reply Program.cs:166]: That's fine.

File: Program.cs [hunk 2]
```diff
@@ -216,5 +220,18 @@ private static ServicePrincipalLoginInformation ParseAuthFile(string authFile)
 
             return info;
         }
+
+        private static ClientSecretCredential ParseAuthFileToCredential(string authFile)
```
Comment (jongio) [inline Program.cs:224]: You can refactor this method with the previous ParseAuthFile - one can call the other.

File: README.md [hunk 3]
```diff
@@ -1,43 +1,86 @@
...
 products:
-- azure
-extensions:
-- services: App-Service
-- platforms: dotnet
+  - azure
+  - dotnet
+  - azure-app-service
+  - azure-key-vault
+  - azure-cosmosDB
```
Comment (jongio) [inline README.md:10]: Make sure casing is right.  I think this has to be lower

File: README.md [hunk 4]
```diff
@@ -1,43 +1,86 @@
...
+description: "This sample show how to use Azure Key Vault store Cosmos DB credential, and then create a web app interacts with the Cosmos DB."
```
Comment (jongio) [inline README.md:11]:
```suggestion
description: "This sample shows how to use Azure Key Vault to store Cosmos DB key, and then create a web app that interacts with the Cosmos DB."
```

File: README.md [hunk 5]
```diff
@@ -1,43 +1,86 @@
...
-      The source code of the web app is located at Asset/documentdb-dotnet-todo-app
+## Prerequisites
+
+To complete this tutorial:
+
+If you don't have an Azure subscription, create a [free account] before you begin.
+
+### Create an auth file
+
+This project requires a auth file be stored in an environment variable securely on the machine running the sample. You can generate this file using Azure CLI 2.0 through the following command. Make sure you selected your subscription by az account set --subscription <name or id> and you have the privileges to create service principals.
```
Comment (jongio) [inline README.md:25]:
```suggestion
This project requires an auth file path be stored in an environment variable securely on the machine running the sample. You can generate this file using Azure CLI 2.0 through the following command. Make sure you selected your subscription by az account set --subscription <name or id> and you have the privileges to create service principals.
```

File: README.md (hunk 4 above)
Comment (FrankieTF) [reply README.md]: updated

File: README.md (hunk 5 above)
Comment (FrankieTF) [reply README.md]: updated

File: README.md (hunk 3 above)
Comment (FrankieTF) [reply README.md]: updated

File: Program.cs (hunk 2 above)
Comment (FrankieTF) [reply Program.cs]: updated

File: Common/Utilities.cs [hunk 6]
```diff
@@ -1,53 +1,48 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using Azure.Messaging.ServiceBus;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using CoreFtp;
```
Comment (jongio) [inline Common/Utilities.cs:7]: Is CoreFtp used in this sample?

File: Common/Utilities.cs [hunk 7]
```diff
@@ -3114,9 +3108,10 @@ public static void SendMessageToTopic(string connectionString, string topicName,
             {
                 try
                 {
-                    var topicClient = new TopicClient(connectionString, topicName);
-                    topicClient.SendAsync(new Message(Encoding.UTF8.GetBytes(message))).Wait();
-                    topicClient.Close();
+                    ServiceBusClient client = new ServiceBusClient(connectionString);
+                    var sender = client.CreateSender(topicName);
+
+                    sender.SendAsync(new ServiceBusMessage(Encoding.UTF8.GetBytes(message))).GetAwaiter().GetResult();
```
Comment (jongio) [inline Common/Utilities.cs:3114]: @jsquire Can you please comment on this .GetAwaiter().GetResult() usage.

File: Common/Utilities.cs (hunk 6 above)
Comment (FrankieTF) [reply Common/Utilities.cs:7]: Yes, CoreFtp is used in Utilities.UploadFileToWebApp(), FtpClient.

**Comment by jongio** [comment]
@heaths - Please review

File: Common/Utilities.cs (hunk 7 above)
Comment (FrankieTF) [reply Common/Utilities.cs]: Resolved, Heaths has reviewed this in another PR-https://github.com/Azure-Samples/key-vault-dotnet-manage-key-vaults/pull/8
we decide to use await for async code, followed by .ConfigureAwait(false).

**Pull request merged**